Add ability to view list of items by SRS category

Deadpikle opened the pull request:
You now have the ability to view all apprentice/guru/etc. items at once. Useful especially for burned items.

![Simulator Screenshot - iPhone SE (3rd generation) - 2023-12-27 at 14 42 11](https://github.com/davidsansome/tsurukame/assets/5092399/2fede84f-37f6-4af1-8f70-32f6b75bfd7f)
![Simulator Screenshot - iPhone SE (3rd generation) - 2023-12-27 at 14 43 18](https://github.com/davidsansome/tsurukame/assets/5092399/dae9f453-f1d7-4765-bb45-54ecf51390ba)

davidsansome commented on ios/MainViewController.swift:
```diff
@@ -229,7 +231,14 @@ class MainViewController: UIViewController, LoginViewControllerDelegate,
     for category in SRSStageCategory.apprentice ... SRSStageCategory.burned {
       let count = services.localCachingClient.srsCategoryCounts[category.rawValue]
-      model.add(SRSStageCategoryItem(stageCategory: category, count: Int(count)))
+      let item = SRSStageCategoryItem(stageCategory: category, count: Int(count))
```
Could you set `accessoryType: .disclosureIndicator` if `count > 0`, so it's obvious it can be tapped?

You'll probably have to add it to the `SRSStageCategoryItem` initialiser and pass it through.

davidsansome commented on ios/SubjectsByCategoryViewController.swift:
```diff
@@ -0,0 +1,149 @@
+    navigationItem.title = category.description
+
+    let model = MutableTableModel(tableView: tableView)
+    model.add(section: "Radicals")
```
It looks a bit weird when these section headers are always added even if they're empty. Maybe you could only add the section headers if there are any subjects of that type? See `SubjectsRemainingViewController` for an example.

davidsansome reviewed (changes requested):
Thanks for this change, it looks great! Just a couple of comments!

Deadpikle commented:
Updated based on your review and rebased on latest `master`. ...not sure what happened with the disclosure indicators there; my screenshots have them but clearly they were gone in the code. Musta done something. Thanks for catching that and for reviewing.
